api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/ClaimantCategoryCalculator.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.dhsc.htbhf.claimant.reporting;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PregnancyEntitlementCalculator;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+import uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.List;
+
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.*;
+
+@Component
+@RequiredArgsConstructor
+public class ClaimantCategoryCalculator {
+
+    private static final int SIXTEEN = 16;
+    private static final int EIGHTEEN = 18;
+
+    private final PregnancyEntitlementCalculator pregnancyEntitlementCalculator;
+
+    /**
+     * Determines the {@link uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory} for a given claimant and their children's date of birth.
+     *
+     * @param claimant               the claimant to determine the category of
+     * @param datesOfBirthOfChildren the dates of birth of the claimant's children
+     * @param atDate                 the date to use when checking a claimant/child's age or pregnancy
+     * @return claimant's determined category
+     */
+    public ClaimantCategory determineClaimantCategory(Claimant claimant, List<LocalDate> datesOfBirthOfChildren, LocalDate atDate) {
+        if (isClaimantPregnant(claimant, atDate)) {
+            int claimantAgeInYears = getClaimantAgeInYears(claimant, atDate);
+            if (claimantAgeInYears < SIXTEEN) {
+                return PREGNANT_AND_UNDER_16;
+            }
+            if (claimantAgeInYears < EIGHTEEN) {
+                return PREGNANT_AND_UNDER_18;
+            }
+            if (hasChildren(datesOfBirthOfChildren, atDate)) {
+                return PREGNANT_WITH_CHILDREN;
+            }
+            return PREGNANT_WITH_NO_CHILDREN;
+        }
+        if (hasChildren(datesOfBirthOfChildren, atDate)) {
+            return NOT_PREGNANT_WITH_CHILDREN;
+        }
+        // this could happen once a claim has expired
+        return NOT_PREGNANT_WITH_NO_CHILDREN;
+    }
+
+    private boolean hasChildren(List<LocalDate> datesOfBirthOfChildren, LocalDate atDate) {
```

Comment:
Feels like this should go in `ChildDateOfBirthCalculator` its already doing a similar calculation for a PaymentCycle so might be good to combine with the method in there.

Reply:
Will do in next PR along with some other refactors.